builder/vsphere/common/step_config_params.go

```diff
@@ -51,12 +51,12 @@ func (s *StepConfigParams) Run(_ context.Context, state multistep.StateBag) mult
 		if s.Config.ToolsUpgradePolicy {
 			info.ToolsUpgradePolicy = "UpgradeAtPowerCycle"
 		}
+	}
 
-		ui.Say("Adding configuration parameters...")
-		if err := vm.AddConfigParams(configParams, info); err != nil {
-			state.Put("error", fmt.Errorf("error adding configuration parameters: %v", err))
-			return multistep.ActionHalt
-		}
+	ui.Say("Adding configuration parameters...")
+	if err := vm.AddConfigParams(configParams, info); err != nil {
```

Comment:
Is there a reason why `vm.AddConfigParams` couldn't be called within the `s.Config.ConfigParams` if block or after line #41? 

I think keeping the variable initialization and usage close to each can help with readability.

Reply:
there're two types of config the `info` and `configParams`, they are both belonging to the same struct when sending this information to the VM. One can set both or only one of them. 
The if statement where the `vm.AddConfigParams` was, was for writing the `info` object, so this was being called only when `info` options were set. 
I remember I was the one that reviewed the PR that introduced this bug and I didn't pay attention to this 🤦

Reply:
Ah so subtle :laughing:

I saw the three dots above and thought configrParams was being unpacked. Thanks for the context.

Reply:
no problem!! :D